scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,882 @@
+------------------------------------
+-- Egg Hunt Egg-Stravaganza
+------------------------------------
+require("scripts/globals/settings")
+require("scripts/globals/zone")
+require("scripts/globals/utils")
+require("scripts/globals/npc_util")
+------------------------------------
+xi = xi or {}
+xi.events = xi.events or {}
+xi.events.egg_hunt = xi.events.egg_hunt or {}
+xi.events.egg_hunt.data = xi.events.egg_hunt.data or {}
+xi.events.egg_hunt.entities = xi.events.egg_hunt.entities or {}
+
+local event = SeasonalEvent:new("egg_hunt")
+
+-- Default settings
+local settings =
+{
+    START  = { DAY   =  6, MONTH = 4 },
+    FINISH = { DAY   = 17, MONTH = 4 },
+
+    VAR =
+    {
+        DAILY_EGG     = "[EGG_HUNT]DAILY_EGG",
+        DAILY_REWARD  = "[EGG_HUNT]DAILY_REWARD",
+        DAILY_BONUS   = "[EGG_HUNT]DAILY_BONUS",
+        DAILY_HELM    = "[EGG_HUNT]DAILY_HELM",
+        FIRST_THREE   = "[EGG_HUNT]FIRST_THREE",
+    },
+
+    -- Default era is 2005
+    ERA_2006 = false, -- Orphic Egg
+    ERA_2007 = false, -- Jeweled Egg and Egg Helm
+    ERA_2008 = false, -- Tier 2 nation eggs, allows trading Hard-Boiled Eggs
+    ERA_2009 = false, -- Egg Buffet set
+    -- 2009, 2010, 2011 and 2012 are identical
+    ERA_2013 = false, -- Prinseggstarta
+    ERA_2014 = false, -- Hatchling Shield, Copse Candy, Cracker
+    ERA_2015 = false, -- Rabbit Cap, show Rabbit Cap wearing NPCs
+    ERA_2018 = false, -- Allows trading Sairui-Ran x99 and Imperial Egg
+    ERA_2019 = false, -- Allows trading Apkallu Egg
+
+    -- Consolation prizes for repeating combinations where
+    -- the player has already received the relevant reward
+    MINOR_REWARDS = true,
+
+    -- Don't add words here
+    -- settings/main.lua will overwrite these
+    BONUS_WORDS = {},
+}
+
+local function loadSettings(tbl, name)
+    if not xi.settings.main[name] then
+        return
+    end
+
+    for k, v in pairs(tbl) do
+        if xi.settings.main[name][k] then
+            tbl[k] = xi.settings.main[name][k]
+        end
+    end
+end
+
+loadSettings(settings, "EGG_HUNT")
+
+xi.events.egg_hunt.enabledCheck = function()
+    local month = tonumber(os.date("%m"))
+    local day = tonumber(os.date("%d"))
+
+    if month == settings.START.MONTH then
+        if day >= settings.START.DAY then
+            return true
+        end
+
+    elseif month == settings.FINISH.MONTH then
+        if day <= settings.FINISH.DAY then
+            return true
+        end
+    end
+
+    return false
+end
```

Comment:
This is good, might look at standardising this across other events

Reply:
I like that its local to the event's own file

Reply:
If we get more seasonal events coded, later we could have a single event settings file.